File: sfas_mobile/lib/features/items/services/item_service.dart
import 'package:dio/dio.dart';
import '../../../core/constants/endpoints.dart';
import '../models/item.dart';

class ItemService {
  final Dio _dio;

  ItemService(this._dio);

  Future<List<Item>> getItems() async {
    try {
      final response = await _dio.get(Endpoints.items);
      final List<dynamic> data = response.data;
      return data.map((json) => Item.fromJson(json)).toList();
    } catch (e) {
      rethrow;
    }
  }

  Future<Item> createItem(Item item) async {
    try {
      final response = await _dio.post(
        Endpoints.items,
        data: item.toJson(),
      );
      return Item.fromJson(response.data);
    } catch (e) {
      rethrow;
    }
  }

  Future<Item> updateItem(String id, Item item) async {
    try {
      final response = await _dio.put(
        Endpoints.itemDetails(id),
        data: item.toJson(),
      );
      return Item.fromJson(response.data);
    } catch (e) {
      rethrow;
    }
  }

  Future<void> deleteItem(String id) async {
    try {
      await _dio.delete(Endpoints.itemDetails(id));
    } catch (e) {
      rethrow;
    }
  }
}

File: sfas-ui/src/app/core/services/item.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Item, ItemRequest } from '../models/item.model';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ItemService {
  private http = inject(HttpClient);
  private baseUrl = environment.apiUrl + '/api/v1/items';

  getAllItems(): Observable<Item[]> {
    return this.http.get<Item[]>(this.baseUrl);
  }

  getItemById(id: string): Observable<Item> {
    return this.http.get<Item>(`${this.baseUrl}/${id}`);
  }

  createItem(item: ItemRequest): Observable<Item> {
    return this.http.post<Item>(this.baseUrl, item);
  }

  updateItem(id: string, item: ItemRequest): Observable<Item> {
    return this.http.put<Item>(`${this.baseUrl}/${id}`, item);
  }

  deleteItem(id: string): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
